github/src/main/kotlin/gropius/sync/github/IssuePile.kt

```diff
@@ -0,0 +1,713 @@
+            }
+
+            is ClosedEventTimelineItemData -> ClosedEventTimelineItem(data)
+            is ReopenedEventTimelineItemData -> ReopenedEventTimelineItem(data)
+            is LabeledEventTimelineItemData -> LabeledEventTimelineItem(data)
+            is UnlabeledEventTimelineItemData -> UnlabeledEventTimelineItem(data)
+            is RenamedTitleEventTimelineItemData -> RenamedTitleEventTimelineItem(data)
+            is AssignedEventTimelineItemData -> AssignedTimelineItem(data)
+            is UnassignedEventTimelineItemData -> UnassignedTimelineItem(data)
+
+            // Handle all events separately, as GitHub does not inherit the createdAt time anywhere
+            is CommentDeletedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is DemilestonedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is MarkedAsDuplicateEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is MentionedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is MilestonedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is PinnedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is UnmarkedAsDuplicateEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is UnpinnedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+
+            is AddedToProjectEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is ConnectedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is ConvertedNoteToIssueEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is ConvertedToDiscussionEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is CrossReferencedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is DisconnectedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is LockedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is MovedColumnsInProjectEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is ReferencedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is RemovedFromProjectEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is TransferredEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is UnlockedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is SubscribedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            is UnsubscribedEventTimelineItemData -> UnknownTimelineItem(data.id, data.createdAt)
+            else -> {
+                //throw IllegalArgumentException("Invalid GraphQL query response in timeline")
```

Comment:
commented out code

Reply:
we want to keep this ready, as we are unsure about the long-term practical handling on unsynced items at the current time

Reply:
then add a TODO